### PR TITLE
feat: added Hive Reader and Writer

### PR DESCRIPTION
**Pull request by SergeyAbzyanov**
Added HiveReader, HiveWriter. Update dependencies

### CODE REVIEW

File: blocks/hive/sources.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        raise TypeError('Not valid codec type: {0}'.format(type(query.codec)))
+    row_codec = query.codec if issubclass(query.codec, Row) else query.codec.type()
+    with conn.cursor() as cur:
+        logger.info(f'Executing query: {query.text}')
+        cur.execute(query.text)
+        logger.info('Validating')
+
+        # Each column in description returns Tuple where first value is column name.
+        # Column name contains table name - remove it for validation
+        keys = [col[0].replace(f'{table_name}.', '') for col in cur.description]
```
Comment (tribunsky-kir) [inline blocks/hive/sources.py:25]: May cause bugs when table name and column name are the same for some reason. 

Maybe it's better to split via '.' or parse in any other way?

File: blocks/hive/processors.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+from blocks.logger import logger
+from blocks.db.types import Query, Row, Table
+
+from pyhive.hive import Connection
+
+
+def prepare_query_value(value: Union[int, float, str]) -> str:
+    new_value = str(value)
+    if isinstance(value, str):
+        new_value = "'{0}'".format(new_value)
```
[the remark this answers is not in the document]
Comment (SergeyAbzyanov) [reply blocks/hive/processors.py]: fixed

File: blocks/hive/sources.py (hunk 1 above)
Comment (SergeyAbzyanov) [reply blocks/hive/sources.py]: Good point. Fixed.

File: blocks/hive/processors.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+def _exec_queries(conn: Connection, query_text: str, partition_key: Union[str, None], rows: Table) -> None:
+    total_count = len(rows.values)
+    with conn.cursor() as cursor:
+        for ix, row in enumerate(rows.rows, 1):
+            row_dict = row.as_dict
+            query = query_text
+            query_args: List[str] = []
+
+            if 'partition' in query_text.lower():
```
Comment (tribunsky-kir) [inline blocks/hive/processors.py:18]: Possible bug if some column or table has 'partition' in it's name.

Comment (SergeyAbzyanov) [reply blocks/hive/processors.py]: Agreed regarding table. Fixed
Regarding columns - column names will be added after this check so there shouldn't be issue with that.

**Review by tribunsky-kir — APPROVED** [review]

**Pull request merged**